src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -35,7 +35,8 @@ public override void Initialize(IHostContext hostContext)
         {
             base.Initialize(hostContext);
             _dockerManger = HostContext.GetService<IDockerCommandManager>();
-            _containerNetwork = $"vsts_network_{Guid.NewGuid().ToString("N")}";
+            var useHostNetwork = AgentKnobs.DockerNetworkCreateDriver.GetValue(hostContext).AsString() == "host";
```

Comment:
I'm not sure about this line: whether `AgentKnobs.DockerNetworkCreateDriver.GetValue()` can get the value from an `IHostContext `.